### PR TITLE
TEST-#1628: Test some methods of categorical dataframes and series.

### LINKED ISSUE #1628: Creating Series with category dtype leads to inconsistent object
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 1809
- **Modin version** (`modin.__version__`): 0.7.3+129.g793eafd.dirty
- **Python version**: 3.7.5
- **Code we can use to reproduce**:
```python
if __name__ == "__main__":
    import numpy as np
    import modin.pandas as pd

    kwargs = {
        "data": np.arange(256),
        "dtype": "category",
    }

    md_series = pd.Series(**kwargs)
    print(md_series.max())

```

### Describe the problem
After that way of creating modin Series (with category dtype), its behavior is unpredictable. Some operations may give incorrect answers (for ex. `max`), others would throw an exception (for ex. `to_pandas`), another would raise an internal error (for ex. `to_numpy`).

The output of related code will be:
```
223
```
But it's obvious that it must be a `255`.
### Source code / logs
If we would print content of Series partitions, we will see that index and values lengths of `__reduced__` column is different:
```
>>> md_series._query_compiler._modin_frame._partitions[0][0].to_pandas()
   __reduced__
0            0
1            1
2            2
3            3
4            4
..         ...
59          27
60          28
61          29
62          30
63          31

[64 rows x 1 columns]
>>> len(md_series._query_compiler._modin_frame._partitions[0][0].to_pandas()["__reduced__"].values)
32
>>> len(md_series._query_compiler._modin_frame._partitions[0][0].to_pandas()["__reduced__"].index)
64
```
So, the partition contains invalid pandas DataFrame. It seems that the problem is in a way how we splitting pandas DataFrame into partitions:
https://github.com/modin-project/modin/blob/e6e24496becb6bb9ddf4dfb9d10448d1a799d459/modin/engines/base/frame/partition_manager.py#L291-L301
The thing is, that when pandas DataFrame contains only one column and its `dtype=category` (this is the case, when we're creating modin Series with category dtype) then `iloc` slicing behaves strange.
Example:
```
>>> import pandas
>>> pandas.__version__
'1.0.3'
>>> df = pandas.DataFrame([1,2,3,4,5], dtype="category")
>>> df.iloc[0:5, 0:2]
   0
0  1
1  2
2
3
4
>>> len(df.iloc[0:5, 0:2].index)
5
>>> len(df.iloc[0:5, 0:2].values)
2
```
Not sure if this a documented behavior or there is a bug in Pandas, but that seems to be causing the problem in modin also

### PR DESCRIPTION
**Pull request by mvashishtha**
In #1628, some functions didn't work on Series with category dtype. Test some
other functions that Categorical natively supports.

Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

In #1628, some functions didn't work on Series with category dtype. Test some
other functions that Categorical natively supports.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1628 
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/utils.py [hunk 1]
```diff
@@ -203,6 +203,11 @@
 test_data_categorical_values = list(test_data_categorical.values())
 test_data_categorical_keys = list(test_data_categorical.keys())
 
+test_data_categorical_dataframe = {
+    "A": pandas.Categorical([1, np.NaN]),
+    "B": pandas.Categorical([np.NaN, 2]),
```
Comment (dchigarev) [inline modin/pandas/test/utils.py:208]: the essential problem of #1628 was that categorical data couldn't be split into partitions properly, let's the amount of rows, so they won't fit into a single partition. Default rows size in tests is 256, with this size they fully fill all of the 4 partitions used in tests.

Comment (mvashishtha) [reply modin/pandas/test/utils.py:208]: Done.

File: modin/pandas/test/utils.py [hunk 2]
```diff
@@ -203,6 +204,16 @@
 test_data_categorical_values = list(test_data_categorical.values())
 test_data_categorical_keys = list(test_data_categorical.keys())
 
+# Fully fill all of the partitions used in tests.
+test_data_large_categorical_dataframe = {
+    i: pandas.Categorical([0] * (NPartitions.get() * 32))
```
Comment (vnlitvinov) [inline modin/pandas/test/utils.py:209]: note that this does not represent the original reproducer - in the #1628 all the category entries were distinct while here you create a "NPartitions * 32" entries of a single category

**Comment by vnlitvinov** [comment]
@mvashishtha please update PR title so it doesn't end in ellipsis, otherwise it would look very strange in short `git log` (and it makes composing release notes harder).

File: modin/pandas/test/utils.py (hunk 2 above)
Comment (mvashishtha) [reply modin/pandas/test/utils.py]: Fixed.

**Comment by mvashishtha** [comment]
> @mvashishtha please update PR title so it doesn't end in ellipsis, otherwise it would look very strange in short `git log` (and it makes composing release notes harder).

Done. For future reference, the cutoff length for PR titles in GitHub appears to be only 69 characters.

**Review by dchigarev — APPROVED** [review]
LGTM

**Review by devin-petersohn — APPROVED** [review]
LGTM! Thanks @mvashishtha

**Comment by vnlitvinov** [comment]
> Done. For future reference, the cutoff length for PR titles in GitHub appears to be only 69 characters.

I wonder if we should limit the commitlint check with 69 then... thoughts @modin-project/modin-core ?

File: modin/pandas/test/utils.py [hunk 3]
```diff
@@ -203,6 +204,16 @@
 test_data_categorical_values = list(test_data_categorical.values())
 test_data_categorical_keys = list(test_data_categorical.keys())
 
+# Fully fill all of the partitions used in tests.
+test_data_large_categorical_dataframe = {
+    i: pandas.Categorical(np.arange(NPartitions.get() * 32))
+    for i in range(NPartitions.get() * 32)
```
Comment (YarShev) [inline modin/pandas/test/utils.py:210]: We should have a variable for minimal partition size.

Comment (mvashishtha) [reply modin/pandas/test/utils.py:210]: I created https://github.com/modin-project/modin/pull/3751 for that.

Comment (YarShev) [reply modin/pandas/test/utils.py:210]: @mvashishtha , thanks!

**Comment by devin-petersohn** [comment]
>I wonder if we should limit the commitlint check with 69 then... thoughts @modin-project/modin-core ?

It started as that, but then there were issues. I think we can just leave it.

**Pull request merged**